src/pybind/mgr/volumes/fs/stats_util.py

```diff
@@ -111,10 +112,22 @@ def __init__(self, volclient, vol_spec):
         # progress event ID for ongoing+pending clone jobs
         self.onpen_pev_id: Optional[str] = 'mgr-vol-total-clones'
 
+        # Make RTimer thread wait for new clones for a minute before initiating
+        # the finish procedure. If a new clone is launched during that period
+        # (which will cause initiate_reporting() to be called), set the
+        # internal flag in this Event. Then, finish procedure won't be invoked
+        # so that RTimer thread can continue reporting progress of cloning.
+        self.wait_before_finish = Event()
+        self.wait_before_finish.clear()
+
     def initiate_reporting(self):
         if self.update_task.is_alive():
             log.info('progress reporting thread is already alive, not '
                      'initiating it again')
+
+            if not self.wait_before_finish.is_set():
```

Comment:
nit: just call set, maybe?

Reply:
I prefer current since it indicates that we expect it to be unset.